docs/contributing/writing-instrumentation-module.md

```diff
@@ -360,3 +360,129 @@ For example:
 ```
 
 [suppress]: https://opentelemetry.io/docs/instrumentation/java/automatic/agent-config/#suppressing-specific-auto-instrumentation
+
+## Use non-inlined advice code with `invokedynamic`
+
+Using non-inlined advice code is possible thanks to the `invokedynamic` instruction, this strategy
+is referred as "indy" in reference to this, by extension "indy modules" are the instrumentation
+modules using this instrumentation strategy.
+
+The most common way to instrument code with bytebuddy relies on inlining, this strategy will be
```

Comment:
Maybe a paragraph here explaining the difference would be nice? E.g.

```
For inlined advices, the advice code is directly copied into the instrumented method. In addition, all helper classes are injected into the classloader of the instrumented classes. For indy, the Advice class is not inlined. Instead, it is loaded alongside with all helper classes into a special `InstrumentationModuleClassloader`, which sees the classes from both the instrumented application classloader and the agent classloader. The instrumented class calls the Advice-class residing in the `InstrumentationModuleClassloader` via invokedynamic bytecode instructions.
```

Reply:
Good idea, I've copied it almost as-is in c477985594d6f8dfb85ff0fce021a2cb47b4bcae